homeassistant/helpers/selector.py

```diff
@@ -131,6 +131,19 @@ def _validate_supported_features(supported_features: int | list[str]) -> int:
     return feature_mask
 
 
+BASE_SELECTOR_CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Optional("read_only"): bool,
+    }
+)
+
+
+class BaseSelectorConfig(TypedDict, total=False):
+    """Class to common options of all selectors."""
+
+    read_only: bool
+
+
 ENTITY_FILTER_SELECTOR_CONFIG_SCHEMA = vol.Schema(
```

Comment:
Is there a reason we don't update `ENTITY_FILTER_SELECTOR_CONFIG_SCHEMA`?

Reply:
ENTITY_FILTER is not a selector. It's a class from which other selectors are derived from. 

To add read_only here would also add it to all the children of EntitySelector.filters, where it wouldn't make sense to exist. 

Probably due to the way that entity filter has multiple legacy formats.

See entity selector definition:

```

class EntitySelectorConfig(BaseSelectorConfig, EntityFilterSelectorConfig, total=False):
    """Class to represent an entity selector config."""

    exclude_entities: list[str]
    include_entities: list[str]
    multiple: bool
    filter: EntityFilterSelectorConfig | list[EntityFilterSelectorConfig]
 ```